### PR TITLE
watcher: major refactor to use `Maybe` types for better error reporting

### PR DESCRIPTION
**Pull request by paperclover**
Previously:

image

This was because:

1. Watchpack is watching files it honestly doesnt need to watch.
2. They check `.code` and ignore some error codes

This PR implements .code for these errors.

### CODE REVIEW

**Comment by paperclover** [comment]
Draft initially to see if anything obvious is caught / improved in CI. will mark as reviewable once I think it is reviewable

File: src/bun.js/node/node_fs_watcher.zig [hunk 1]
```diff
@@ -240,76 +266,56 @@ pub const FSWatcher = struct {
...
         if (!this.refTask()) {
             return;
         }
 
-        const encoding = this.encoding;
-        switch (event_type) {
-            inline .rename, .change => |eventType| {
-                const task = FSWatchTaskWindows.new(.{
-                    .ctx = this,
-                    .file_path = switch (encoding) {
```
Comment (Jarred-Sumner) [inline src/bun.js/node/node_fs_watcher.zig:294]: add back optimization

**Comment by Jarred-Sumner** [comment]
does not compile

**Comment by paperclover** [comment]
regresses fs watch tests on windows

**Pull request merged**